lightning/src/routing/scoring.rs

```diff
@@ -798,6 +797,122 @@ impl<G: Deref<Target = NetworkGraph>, T: Time> Score for ProbabilisticScorerUsin
 	}
 }
 
+mod approx {
+	const BITS: u32 = 64;
+	const HIGHEST_BIT: u32 = BITS - 1;
+	const LOWER_BITS: u32 = 4;
+	const LOWER_BITS_BOUND: u64 = 1 << LOWER_BITS;
+	const LOWER_BITMASK: u64 = (1 << LOWER_BITS) - 1;
+
+	/// Look-up table for `log10(x) * 1024` where row `i` is used for each `x` having `i` as the
+	/// most significant bit. The next 4 bits of `x`, if applicable, are used for the second index.
+	const LOG10_TIMES_1024: [[u16; LOWER_BITS_BOUND as usize]; BITS as usize] = [
+		[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
```

Comment:
Technically we could get rid of the first 3 rows, right? Cause the first row is just representing `1`, the second row `2` and `3`, the third row... I don't care if you bother with it, though, up to you.

Reply:
Yeah, though I wasn't sure if it was worth adding branching for those cases. Keeping it as it is also simplifies the indexing.